### DISCUSSION TITLE
Using a shared folder for sqlite database causing a panic.

### DISCUSSION THREAD

**Discussion by sylvesterroos** [dani-garcia/vaultwarden#1864] [Q&A]
### Subject of the issue
Launching the vaultwarden container immediately crashes with the error message in the title.

### Deployment environment

* Install method: latest docker image

* Other relevant details: I'm running the latest image on a raspberry pi running Ubuntu server 20.04 LTS.

### Steps to reproduce
I set the data directory to be on my (which still has space for files) external hard drive.
`docker run -d --name vaultwarden -v /mnt/storage/Appdata/vaultwarden/:/data/ -p 8100:80 vaultwarden/server:latest`

### Expected behaviour
The container should launch without crashing

### Actual behaviour
Crash
```
thread 'main' panicked at 'Failed to turn on WAL: DatabaseError(__Unknown, "disk I/O error")': src/db/mod.rs:308
```

### Troubleshooting data
docker logs vaultwarden:
[vaultwarden.log](https://github.com/dani-garcia/vaultwarden/files/6864394/log.log)

**Comment by RealOrangeOne** [comment]
> disk I/O error

Looks like it's an error with our disk, not vaultwarden.

Are you running other applications off the drive just fine? It's not about free space, it's more likely the drive is failing, corrupting, or a bad cable.

**Comment by BlackDex** [comment]
I think you are using a shared storage like samba/cifs or nfs. For that to work you need to disable WAL. But i would suggest to use a different type of database in that case like postgresql or mariadb.

**Comment by sylvesterroos** [comment]
The storage folder is indeed shared using samba. You're saying that not sharing this folder will fix the issue?

**Comment by BlackDex** [comment]
Not using a samba/cifs or nfs folder will solve your issue.
